src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java

```diff
@@ -420,4 +451,88 @@ public static boolean isCorrectJavadocPosition(DetailAST blockComment) {
                 || BlockCommentPosition.isOnEnumConstant(blockComment)
                 || BlockCommentPosition.isOnAnnotationDef(blockComment);
     }
+
+    /**
+     * Checks if the supplied {@code DetailNode} argument is not a tight HTML tag.
+     * @param node the node to check
+     * @return true if the javadoc node is not a tight HTML tag; false otherwise
+     */
+    public static boolean isNonTightHtmlTag(DetailNode node) {
+        final boolean isLooseHtmlTag;
+        isLooseHtmlTag = node.getText().endsWith("TAG_START")
+                && node.getParent().getType() != getHtmlTagParentTypeWhenClosed(node);
```

Comment:
@romani 

We can make ANTLR return boolean flag attached to the HTML tag that is `nonTight`. But that will be attached to ANTLR `ParseTree` and not our `DetailNode` tree. So say we set some boolean somewhere to indicate if a javadoc Tree is nonTight, we can do that. But then the only issue will be to log a detailed message about which tag is nonTight. 

So how to propogate that information ? Create a class to hold all the information about the first nonTight HTML tag encountered and the `AbstractJavadocCheck` can use that object ?